utility/src/test/java/org/oppia/android/util/profile/ProfileNameValidatorTest.kt

```diff
@@ -0,0 +1,119 @@
+package org.oppia.android.util.profile
+
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.testing.junit.OppiaParameterizedTestRunner
+import org.oppia.android.testing.junit.OppiaParameterizedTestRunner.Iteration
+import org.oppia.android.testing.junit.OppiaParameterizedTestRunner.Parameter
+import org.oppia.android.testing.junit.OppiaParameterizedTestRunner.RunParameterized
+import org.oppia.android.testing.junit.OppiaParameterizedTestRunner.SelectRunnerPlatform
+import org.oppia.android.testing.junit.ParameterizedRobolectricTestRunner
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Suppress("FunctionName")
+@RunWith(OppiaParameterizedTestRunner::class)
+@SelectRunnerPlatform(ParameterizedRobolectricTestRunner::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(manifest = Config.NONE)
+class ProfileNameValidatorTest {
+  @Inject
+  lateinit var profileNameValidator: ProfileNameValidator
+
+  @Parameter
+  lateinit var name: String
+
+  @Before
+  fun setup() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testIsNameValid_nameWithSpaces_returnsFalse() {
+    val nameWithSpaces = "Ben Henning"
+    assertThat(profileNameValidator.isNameValid(nameWithSpaces)).isFalse()
+  }
+
+  @Test
+  fun testIsNameValid_nameWithNumber_returnsFalse() {
+    val nameWithNumber = "Jishnu7"
+    assertThat(profileNameValidator.isNameValid(nameWithNumber)).isFalse()
+  }
+
+  @Test
+  @RunParameterized(
+    Iteration("Ben#Henning", "name=Ben#Henning"),
+    Iteration("Rajay@T", "name=Rajay@T"),
+    Iteration("جيشنو^&&", "name=جيشنو^&&"),
+    Iteration("_Jishnu", "name=_Jishnu"),
```

Comment:
Nit: Please remove the trailing comma for here & other parameterized tests below.

Reply:
done